core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -1709,6 +1709,8 @@ public function undo()
 				// Unset fields that no longer exist in the database
 				$row = array_intersect_key($row, $arrFields[$table]);
 
+				$this->denyAccessUnlessGranted(ContaoCorePermissions::DC_PREFIX . $this->strTable, new CreateAction($table, $row));
```

Comment:
the prefix should use `$table` (the table that is restored to) not `$this->strTable`. Also, this would restore partial data if one of the records cannot be written. Maybe we should use a transaction for that, and cancel it if something fails before (re)throwing the exception?

Reply:
Fixed. I think we could but it's unrelated to this PR and could even be considered a bugfix. The same happens now if one insert statement fails.

Reply:
yes, but having a permission not granted is very much more likely than an insert statement to fail, is it not?

Reply:
Yeah but it also does check for some affected rows and deletes it again then magic. I really don't want to touch this here.